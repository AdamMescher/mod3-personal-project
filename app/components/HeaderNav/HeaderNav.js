import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class HeaderNav extends Component {
  render() {
    const { currentUser, usersHouse, signOut, leaveHouse } = this.props;
    const userStuff = <div className='user-stuff'>
      <p className='greeting'>{`Welcome to ${usersHouse.houseName},  ${currentUser.name}.`}</p>
      <button onClick={() => { leaveHouse(currentUser, usersHouse); }}>Leave House</button>
      <button onClick={signOut}>Sign Out</button>
    </div>;
    return (
      <div>
        <Link to="/">
          <h1>Tribe</h1>
        </Link>
        <h3>A Communal Living Management App</h3>
        {usersHouse.houseName ? userStuff : null}
        <div>
          <Link to="/summary">House Summary</Link>
          <Link to="/bulletins">House Bulletins</Link>
          <Link to="/bills">House Bills</Link>
          <Link to="/chores">House Chores</Link>
          <Link to="/bills/user">My Bills</Link>
          <Link to="/chores/user">My Chores</Link>
          <div className='leave-house-div' onClick={() => { leaveHouse(currentUser, usersHouse); }}>Leave House</div>
          <div className='signout-div' onClick={signOut}>Sign Out</div>
        </div>
      </div>
    );
  }
}

export default HeaderNav;

HeaderNav.propTypes = {
  currentUser: PropTypes.object,
  usersHouse: PropTypes.object,
  signOut: PropTypes.func,
  leaveHouse: PropTypes.func
};
