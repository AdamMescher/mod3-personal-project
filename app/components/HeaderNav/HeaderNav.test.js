import React from "react";
import HeaderNav from './HeaderNav.js';
import { shallow } from "enzyme";

describe('HeaderNav', () => {
  const user = {
    id: 'abc',
    name: 'Lola'
  };
  const house = {
    houseKey: 'abc',
    houseName: 'fake house',
    houseCode: '6666',
    users: [user]
  };
  const mockFn = jest.fn();
  const wrapper = shallow(<HeaderNav
    currentUser={user}
    usersHouse={house}
    signOut={mockFn}
    leaveHouse={mockFn}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a title and subtitle', () => {
    const title = wrapper.find('h1');
    const subtitle = wrapper.find('h3');

    expect(title.length).toEqual(1);
    expect(subtitle.length).toEqual(1);
  });

  it('should have seven links', () => {
    const link = wrapper.find('Link');

    expect(link.length).toEqual(7);
  });

  it('should have signout and leavehouse div', () => {
    const signoutDiv = wrapper.find('.signout-div');
    const leavehouseDiv = wrapper.find('.leave-house-div');

    expect(signoutDiv.length).toEqual(1);
    expect(leavehouseDiv.length).toEqual(1);
  });

  it('should render userStuff', () => {
    const userStuff = wrapper.find('.user-stuff');
    const greeting = wrapper.find('.greeting');
    const buttons = wrapper.find('button');

    expect(userStuff.length).toEqual(1);
    expect(greeting.text()).toEqual('Welcome to fake house,  Lola.');
    expect(buttons.length).toEqual(2);
  });

  it('should not render userStuff when no user', () => {
    const altWrapper = shallow(<HeaderNav
      currentUser={{}}
      usersHouse={{}}
      signOut={mockFn}
      leaveHouse={mockFn}/>);
    const userStuff = altWrapper.find('.user-stuff');

    expect(userStuff.length).toEqual(0);
  });

  it('should fire actions', () => {
    const button1 = wrapper.find('button').first();
    const button2 = wrapper.find('button').last();
    const signoutDiv = wrapper.find('.signout-div');
    const leavehouseDiv = wrapper.find('.leave-house-div');

    button1.simulate('click');
    button2.simulate('click');
    signoutDiv.simulate('click');
    leavehouseDiv.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(4);
  });
});
