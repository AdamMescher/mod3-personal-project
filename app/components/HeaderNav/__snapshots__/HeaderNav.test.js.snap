// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderNav
    currentUser={
      Object {
        "id": "abc",
        "name": "Lola",
      }
    }
    leaveHouse={[Function]}
    signOut={[Function]}
    usersHouse={
      Object {
        "houseCode": "6666",
        "houseKey": "abc",
        "houseName": "fake house",
        "users": Array [
          Object {
            "id": "abc",
            "name": "Lola",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          <h1>
            Tribe
          </h1>
        </Link>,
        <h3>
          A Communal Living Management App
        </h3>,
        <div
          className="user-stuff"
        >
          <p
            className="greeting"
          >
            Welcome to fake house,  Lola.
          </p>
          <button
            onClick={[Function]}
          >
            Leave House
          </button>
          <button
            onClick={[Function]}
          >
            Sign Out
          </button>
        </div>,
        <div>
          <Link
            replace={false}
            to="/summary"
          >
            House Summary
          </Link>
          <Link
            replace={false}
            to="/bulletins"
          >
            House Bulletins
          </Link>
          <Link
            replace={false}
            to="/bills"
          >
            House Bills
          </Link>
          <Link
            replace={false}
            to="/chores"
          >
            House Chores
          </Link>
          <Link
            replace={false}
            to="/bills/user"
          >
            My Bills
          </Link>
          <Link
            replace={false}
            to="/chores/user"
          >
            My Chores
          </Link>
          <div
            className="leave-house-div"
            onClick={[Function]}
          >
            Leave House
          </div>
          <div
            className="signout-div"
            onClick={[Function]}
          >
            Sign Out
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            Tribe
          </h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Tribe",
          },
          "ref": null,
          "rendered": "Tribe",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "A Communal Living Management App",
        },
        "ref": null,
        "rendered": "A Communal Living Management App",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="greeting"
            >
              Welcome to fake house,  Lola.
            </p>,
            <button
              onClick={[Function]}
            >
              Leave House
            </button>,
            <button
              onClick={[Function]}
            >
              Sign Out
            </button>,
          ],
          "className": "user-stuff",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to fake house,  Lola.",
              "className": "greeting",
            },
            "ref": null,
            "rendered": "Welcome to fake house,  Lola.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Leave House",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Leave House",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Out",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/summary"
            >
              House Summary
            </Link>,
            <Link
              replace={false}
              to="/bulletins"
            >
              House Bulletins
            </Link>,
            <Link
              replace={false}
              to="/bills"
            >
              House Bills
            </Link>,
            <Link
              replace={false}
              to="/chores"
            >
              House Chores
            </Link>,
            <Link
              replace={false}
              to="/bills/user"
            >
              My Bills
            </Link>,
            <Link
              replace={false}
              to="/chores/user"
            >
              My Chores
            </Link>,
            <div
              className="leave-house-div"
              onClick={[Function]}
            >
              Leave House
            </div>,
            <div
              className="signout-div"
              onClick={[Function]}
            >
              Sign Out
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "House Summary",
              "replace": false,
              "to": "/summary",
            },
            "ref": null,
            "rendered": "House Summary",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "House Bulletins",
              "replace": false,
              "to": "/bulletins",
            },
            "ref": null,
            "rendered": "House Bulletins",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "House Bills",
              "replace": false,
              "to": "/bills",
            },
            "ref": null,
            "rendered": "House Bills",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "House Chores",
              "replace": false,
              "to": "/chores",
            },
            "ref": null,
            "rendered": "House Chores",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "My Bills",
              "replace": false,
              "to": "/bills/user",
            },
            "ref": null,
            "rendered": "My Bills",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "My Chores",
              "replace": false,
              "to": "/chores/user",
            },
            "ref": null,
            "rendered": "My Chores",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Leave House",
              "className": "leave-house-div",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Leave House",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Out",
              "className": "signout-div",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <h1>
              Tribe
            </h1>
          </Link>,
          <h3>
            A Communal Living Management App
          </h3>,
          <div
            className="user-stuff"
          >
            <p
              className="greeting"
            >
              Welcome to fake house,  Lola.
            </p>
            <button
              onClick={[Function]}
            >
              Leave House
            </button>
            <button
              onClick={[Function]}
            >
              Sign Out
            </button>
          </div>,
          <div>
            <Link
              replace={false}
              to="/summary"
            >
              House Summary
            </Link>
            <Link
              replace={false}
              to="/bulletins"
            >
              House Bulletins
            </Link>
            <Link
              replace={false}
              to="/bills"
            >
              House Bills
            </Link>
            <Link
              replace={false}
              to="/chores"
            >
              House Chores
            </Link>
            <Link
              replace={false}
              to="/bills/user"
            >
              My Bills
            </Link>
            <Link
              replace={false}
              to="/chores/user"
            >
              My Chores
            </Link>
            <div
              className="leave-house-div"
              onClick={[Function]}
            >
              Leave House
            </div>
            <div
              className="signout-div"
              onClick={[Function]}
            >
              Sign Out
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              Tribe
            </h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Tribe",
            },
            "ref": null,
            "rendered": "Tribe",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "A Communal Living Management App",
          },
          "ref": null,
          "rendered": "A Communal Living Management App",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="greeting"
              >
                Welcome to fake house,  Lola.
              </p>,
              <button
                onClick={[Function]}
              >
                Leave House
              </button>,
              <button
                onClick={[Function]}
              >
                Sign Out
              </button>,
            ],
            "className": "user-stuff",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to fake house,  Lola.",
                "className": "greeting",
              },
              "ref": null,
              "rendered": "Welcome to fake house,  Lola.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Leave House",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Leave House",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/summary"
              >
                House Summary
              </Link>,
              <Link
                replace={false}
                to="/bulletins"
              >
                House Bulletins
              </Link>,
              <Link
                replace={false}
                to="/bills"
              >
                House Bills
              </Link>,
              <Link
                replace={false}
                to="/chores"
              >
                House Chores
              </Link>,
              <Link
                replace={false}
                to="/bills/user"
              >
                My Bills
              </Link>,
              <Link
                replace={false}
                to="/chores/user"
              >
                My Chores
              </Link>,
              <div
                className="leave-house-div"
                onClick={[Function]}
              >
                Leave House
              </div>,
              <div
                className="signout-div"
                onClick={[Function]}
              >
                Sign Out
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "House Summary",
                "replace": false,
                "to": "/summary",
              },
              "ref": null,
              "rendered": "House Summary",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "House Bulletins",
                "replace": false,
                "to": "/bulletins",
              },
              "ref": null,
              "rendered": "House Bulletins",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "House Bills",
                "replace": false,
                "to": "/bills",
              },
              "ref": null,
              "rendered": "House Bills",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "House Chores",
                "replace": false,
                "to": "/chores",
              },
              "ref": null,
              "rendered": "House Chores",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "My Bills",
                "replace": false,
                "to": "/bills/user",
              },
              "ref": null,
              "rendered": "My Bills",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "My Chores",
                "replace": false,
                "to": "/chores/user",
              },
              "ref": null,
              "rendered": "My Chores",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Leave House",
                "className": "leave-house-div",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Leave House",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "className": "signout-div",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
